File: examples/sharing/share-batch.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Share a document with several people at once, mixing user IDs and emails.
export async function batchShare(client: JsBaoClient, documentId: string) {
  // #region example
  await client.documents.updatePermissions(documentId, {
    permissions: [
      { userId: "user-abc", permission: "read-write" },
      { email: "alice@example.com", permission: "reader" },
      { email: "bob@example.com", permission: "read-write" },
    ],
  });
  // #endregion example
}

File: examples/sharing/share-batch.swift
import JsBaoClient

// Share a document with several people at once, mixing user IDs and emails.
func batchShare(client: JsBaoClient, documentId: String) async throws {
  // #region example
  _ = try await client.documents.updatePermissions(
    documentId: documentId,
    params: .batch([
      PermissionAssignment(userId: "user-abc", permission: "read-write"),
      PermissionAssignment(email: "alice@example.com", permission: "reader"),
      PermissionAssignment(email: "bob@example.com", permission: "read-write"),
    ])
  )
  // #endregion example
}
